addons/mail/static/src/models/message_sender/message_sender.js

```diff
@@ -0,0 +1,272 @@
+                        'g');
+                    htmlString = htmlString.replace(regexp, '$1' + emoji.unicode);
+                }
+            }
+            return htmlString;
+        }
+
+        async _postMailChannel(postData) {
+            let messageId;
+            const command = this.thread.composer._getCommandFromText(postData.body);
+            Object.assign(postData, {
+                subtype_xmlid: 'mail.mt_comment',
+            });
+            if (command) {
+                messageId = await this.async(() => this.env.models['mail.thread'].performRpcExecuteCommand({
+                    channelId: this.thread.id,
+                    command: command.name,
+                    postData,
+                }));
+            } else {
+                messageId = await this.async(() =>
+                    this.env.models['mail.thread'].performRpcMessagePost({
+                        postData,
+                        threadId: this.thread.id,
+                        threadModel: this.thread.model,
+                    })
+                );
+            }
+            return messageId;
+        }
+
+        async _postMailThread(postData) {
+            Object.assign(postData, {
+                subtype_xmlid: this.thread.composer.isLog ? 'mail.mt_note' : 'mail.mt_comment',
+            });
+            if (!this.thread.composer.isLog) {
+                postData.context = {
+                    mail_post_autofollow: true,
+                };
+            }
+            const messageId = await this.async(() =>
+                this.env.models['mail.thread'].performRpcMessagePost({
+                    postData,
+                    threadId: this.thread.id,
+                    threadModel: this.thread.model,
+                })
+            );
+            const [messageData] = await this.async(() => this.env.services.rpc({
+                model: 'mail.message',
+                method: 'message_format',
+                args: [[messageId]],
+            }, { shadow: true }));
+            this.env.models['mail.message'].insert(Object.assign(
+                {},
+                this.env.models['mail.message'].convertData(messageData),
+                {
+                    originThread: [['insert', {
+                        id: this.thread.id,
+                        model: this.thread.model,
+                    }]],
+                })
+            );
+            this.thread.loadNewMessages();
+        }
+
+        /**
+         * Function that process the message stored to be sent.
+         */
+        async _processMessagesToBeSent() {
+            if (this.isSendingMessages) {
+                return;
+            }
+            this.update({
+                isSendingMessages: true,
+            });
+            const messages = this.pendingMessagesToBeSent.filter((message) => !message.hasSendError);
+            const message = messages[0];
+            try {
+                if (message) {
```

Comment:
I think this is related to previous comment, `_processMessagesToBeSent` should probably not be called if there is no message, or at best it should lead to a quick return.

Reply:
I would keep this guard, just to be safe.